src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.cs

```diff
@@ -73,14 +71,12 @@ public bool TryGetValue(LoadTextOptions options, [MaybeNullWhen(false)] out Text
 
         public bool TryGetTextVersion(LoadTextOptions options, out VersionStamp version)
         {
-            if (_initialSourceOrRecoverableText is ITextVersionable textVersionable)
-            {
-                return textVersionable.TryGetTextVersion(options, out version);
-            }
+            if (TryGetInitialSourceOrRecoverableText(out var source, out var recoverableText))
+                return source.TryGetTextVersion(options, out version);
```

Comment:
defer to common helper which either gives the original 'source' or the 'recoverable text'.  if the former wins, we just defer to that to try to get the version.  This also moves to match the pattern of the method above.